components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/IdentityGovernanceServiceImpl.java

```diff
@@ -233,10 +237,51 @@ public ConnectorConfig getConnectorWithConfigs(String tenantDomain,
 
         for (ConnectorConfig connectorConfig : connectorListWithConfigs) {
             if (connectorConfig.getName().equals(connectorName)) {
+                // Should remove this logic eventually.
+                convertPropertyUseNumericToUseAlphaNumeric(connectorName, connectorConfig);
```

Comment:
this should be done at a lower level. Wouldn't this method get called for any governance configuration update? 
Check if we are updating the email otp configs in a more granular level

Reply:
> this should be done at a lower level. Wouldn't this method get called for any governance configuration update? Check if we are updating the email otp configs in a more granular level

The decision to bring this logic here was taken because this covers both the UI and the API level CRUD operations. Add and Update both are being handled here.